compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/diagnostic/BLangDiagnosticLog.java

```diff
@@ -65,40 +66,58 @@ public static BLangDiagnosticLog getInstance(CompilerContext context) {
         return dLogger;
     }
 
+    public static BLangDiagnosticLog getInstance(CompilerContext context, PackageID packageID) {
+        BLangDiagnosticLog dLogger = context.get(DIAGNOSTIC_LOG_KEY);
+        if (dLogger == null) {
+            dLogger = new BLangDiagnosticLog(context);
+        }
+
+        if (packageID != null) {
+            dLogger.setCurrentPackageId(packageID);
+        }
+
+        return dLogger;
+    }
+
+    @Override
+    public void setCurrentPackageId(PackageID packageID) {
+        this.currentPackageId = packageID;
+    }
```

Comment:
I think we don't need this method, if we always pass the package ID when logging. Since we would eventually move to that, this method won't be needed.

Reply:
But we need this for now as we use `error()` , `warning()` and `note()` methods in `BLangDiagnosticLog` class to log diagnostics. Right?

Reply:
Sorry I meant, we don't need the overloaded `getInstance()` method

Reply:
ah :smile: okay